src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/PremisesController.kt

```diff
@@ -3,16 +3,19 @@ package uk.gov.justice.digital.hmpps.approvedpremisesapi.controller
 import org.springframework.http.ResponseEntity
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.PremisesApiDelegate
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.health.api.model.Booking
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.health.api.model.Premises
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.problem.NotFoundProblem
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.PremisesService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.BookingTransformer
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.PremisesTransformer
 import java.util.UUID
 
 @Service
```

Comment:
Looks like you are using the wrong stereotype.  This is a controller isn't it?

Reply:
I think if you use `@RestController` then you don't have to be explicit about the `ResponseEntity` and Spring will wire that up for you. However I suspect your approach here is dictated by the contract first approach? 

I think this is why we don't use contract first - so we can make use of the various Spring annotations to decorate the endpoints. Here's a [recent example of a controller](https://github.com/ministryofjustice/hmpps-nomis-prisoner-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/resource/VisitResource.kt) of ours and you can see we use lots of annotations!

(As a side note I thought contract first was the way to go until I realised all of the limitations. I found the generators to be quite limited, but was hoping they would have matured by now)

Reply:
It's sort-of-a-controller, the actual Controller is in `build/generated/src/main/kotlin/uk/gov/..../api/PremisesApiController.kt` which just delegates to PremisesApiDelegate which this class implements.  So I think as far as the stereotypes go this is either a @Service or perhaps a @Component 

![image](https://user-images.githubusercontent.com/110207532/183933820-69be3b93-aeff-4b34-a62a-12f4a2eebfe8.png)

```
However I suspect your approach here is dictated by the contract first approach?
```
That's correct, hands are a little tied on this one.  Do you two think that the contract-first approach deviates too much from the norm at MoJ?

```
(As a side note I thought contract first was the way to go until I realised all of the limitations. I found the generators to be quite limited, but was hoping they would have matured by now)
```

We used this a lot on a project at a previous work place - one big issue I found with it was lack of support for oneOf on response bodies when wanting to return entirely distinct responses for failure/error.  We've sidestepped that one here by using the problem-spring-web library.  Were there any other gotchas you remember?